File: programs/valid-paranthese.cs
public class Solution {
    public bool IsValid(string s) {
        Dictionary<char,char> mydict = new Dictionary<char,char> {
            {']','['},
            {')','('},
            {'}','{'}
        };
        
        Stack<char> mystack = new Stack<char>();

        foreach(char c in s){
            if(mydict.ContainsKey(c)){
                if(mystack.Count != 0 && mystack.Peek() == mydict[c]){
                    mystack.Pop();
                }else{
                    return false;
                }
            }else{
                mystack.Push(c);
            }
        }

        if(mystack.Count == 0){
            return true;
        }
        return false;
    }
}

File: programs in python/valid-paranthese.py
class Solution:
    def isValid(self, s: str) -> bool:
        mydict = {
            ")" : "(",
            "]" : "[",
            "}" : "{"
        }

        stack = []

        for c in s:
            if c in mydict:
                if stack and stack[-1] == mydict[c]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(c)
        
        return True if not stack else False